### PR TITLE
- Laravel Dusk configuration and example tests

### PR DESCRIPTION
**Pull request by plencka**
An attempt at making the Dusk work with the current environment, without changing the project structure as previously in the `tests` branch.

- This pull requests contains dusk configuration and a few example tests that should success.
It should allow browser tests to run after creating new docker images and applying instructions.
- There are also some fixes with pagination and updated dependencies.
- I also removed "Manage companies" and moved its functions to search filters to simplify controller.

Instructions are in the testing.md in the project documentation.

### CODE REVIEW

File: backend/tests/Browser/Pages/VerifyEmailPage.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        return "/verify-email";
+    }
+
+    public function elements(): array
+    {
+        return [
+            "@element" => "#selector",
+        ];
+    }
```
Comment (EwelinaLasowy) [inline backend/tests/Browser/Pages/VerifyEmailPage.php:19]:
```suggestion
```

If you don't specify selectors here, you don't need this method.

**Review by EwelinaLasowy — COMMENTED** [review]
1. End to end tests should test some user path - mostly it's happy path, because e2e tests are time-consuming, so this unhappy path we should catch in feature tests (or unit tests). For learning purposes this tests are okay, I won't nit-pick. But in the future, keep that in mind. :wink:
2. How much time is needed for the tests to run? On my computer it takes more than two and a half minutes.
image

File: backend/tests/Browser/CreateCompanyTest.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            $this->getFirstDepartmentFromInputComponent($browser);
+            $browser->press("@form_submit")
+                ->waitUntilMissing("#nprogress")
+                ->assertSee("TestCompanyName");
+
+            $browser->logout();
+        });
+    }
+
+    protected function getFirstDepartmentFromInputComponent(Browser $browser): void
```
Comment (EwelinaLasowy) [inline backend/tests/Browser/CreateCompanyTest.php:68]: Think of making component for this select - https://laravel.com/docs/9.x/dusk#components

**Comment by plencka** [comment]
> 1. End to end tests should test some user path - mostly it's happy path, because e2e tests are time-consuming, so this unhappy path we should catch in feature tests (or unit tests). For learning purposes this tests are okay, I won't nit-pick. But in the future, keep that in mind. 😉
> 2. How much time is needed for the tests to run? On my computer it takes more than two and a half minutes.
>    image
_Originally posted by @EwelinaLasowy in https://github.com/blumilksoftware/internships/pull/198#pullrequestreview-1243423338_

1. Alright. For now I'll leave them as they are.
2. For my setup, it's only ~30 seconds. It's probably related to CPU and the WSL2 configuration, as I'm running the application on Windows 10. I modified the WSL2 to use more memory and cores to fix some other performance issues, so it might be related.

- I've tried finding a solution to optimize tests, but most suggestions recommend modifying the behaviour of database migrations, but it introduces some other issues and requires tests to be written in a certain way.
Tests also seem to be slowed down due to waits, as each one of them adds up. Not having them can cause tests to fail for unloaded elements.
I'm not sure for now what would be the best approach. Some people were complaining that Dusk is slow in general.
- I was not aware about Components. I made a TreeSelect one and the test appears to be cleaner now. :smile:

**Review by EwelinaLasowy — APPROVED** [review]
1. Now tests are faster - it takes almost 43 seconds
image

2. With components this test looks better :)

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**